onnxruntime/core/session/inference_session.cc

```diff
@@ -106,6 +106,8 @@ InferenceSession::InferenceSession(const SessionOptions& session_options, loggin
     int pool_size = session_options_.session_thread_pool_size <= 0
                         ? std::thread::hardware_concurrency() / 2
                         : session_options_.session_thread_pool_size;
+    if (pool_size == 0)
+      pool_size = 1;
 
     thread_pool_ = std::make_unique<onnxruntime::concurrency::ThreadPool>("SESSION", pool_size);
```

Comment:
It would be better if class ThreadPool's constructor also have a check like this.

Reply:
If we add a check in ThreadPool's constructor, do we still need the check here (i.e. inference_session.cc), which would become redundant.

Reply:
ThreadPool should throw exception if any input argument is invalid. I would think zero is not a valid value for thread count.

Reply:
OK. PR updated.